src/lib/connect/tls.ts

```diff
@@ -10,16 +10,20 @@ const debug = _debug('mqttjs:tls')
 function connect(opts: IClientOptions): TLSSocket {
 	const { host, port, socksProxy, ...rest } = opts
 
-	return tls.connect(
-		socksProxy
-			? {
-					...rest,
-					socket: openSocks(host, port, socksProxy, {
-						timeout: opts.socksTimeout,
-					}),
-				}
-			: opts,
-	)
+	if (socksProxy !== undefined) {
+		const socket = openSocks(host, port, socksProxy, {
+			timeout: opts.socksTimeout,
+		})
+
+		// @ts-expect-error - There is a type overlap between readable-stream and stream which are used for browsers and node respectively.
+		// We would need to make it clearer what is aimed at browsers and what is aimed at Node.js to resolve this type issue.
```

Comment:
It would be very beneficial to make it clear which files are aimed at browsers and which files are aimed at Node.js within the project architecture. As it is today I find it too risky to change any interface since there are so many overlaps.